pyintegration/deephaven2/calendars.py

```diff
@@ -0,0 +1,372 @@
+        self.j_business_period = j_business_period
+
+    def __repr__(self):
+        return f"[{self.start_time}, {self.end_time}]"
+
+    @property
+    def j_object(self) -> jpy.JType:
+        return self.j_business_period
+
+    @property
+    def start_time(self) -> DateTime:
+        """ The start of the period. """
+        return self.j_business_period.getStartTime()
+
+    @property
+    def end_time(self) -> DateTime:
+        """ The end of the period. """
+        return self.j_business_period.getEndTime()
+
+    @property
+    def length(self) -> int:
+        """ The length of the period in nanoseconds. """
+        return self.j_business_period.getLength()
+
+
+class BusinessSchedule(JObjectWrapper):
+    """ A business schedule defines a single business day. """
+    j_object_type = _JBusinessSchedule
+
+    def __init__(self, j_business_schedule):
+        self.j_business_schedule = j_business_schedule
+
+    @property
+    def j_object(self) -> jpy.JType:
+        return self.j_business_schedule
+
+    @property
+    def business_periods(self) -> List[BusinessPeriod]:
+        """ The business periods of the day.
+
+        Returns:
+             List[BusinessPeriod]
+        """
+        j_periods = self.j_business_schedule.getBusinessPeriods()
+        periods = []
+        for j_period in j_periods:
+            periods.append(BusinessPeriod(j_period))
+        return periods
+
+    @property
+    def start_of_day(self) -> DateTime:
+        """ The start of the business day. """
+        return self.j_business_schedule.getStartOfBusinessDay()
+
+    @property
+    def end_of_day(self) -> DateTime:
+        """ The end of the business day. """
+        return self.j_business_schedule.getEndOfBusinessDay()
+
+    def is_business_day(self) -> bool:
+        """ Whether it is a business day.
+
+        Returns:
+            bool
+        """
+        return self.j_business_schedule.isBusinessDay()
+
+    def is_business_time(self, time: DateTime) -> bool:
+        """ Whether the specified time is a business time for the day.
+
+        Args:
+             time (DateTime): the time during the day
+
+        Return:
+            bool
+        """
+        return self.j_business_schedule.isBusinessTime(time)
+
+    def business_time_elapsed(self, time: DateTime) -> int:
+        """ Returns the amount of business time in nanoseconds that has elapsed on the given day by the specified
+        time.
+
+        Args:
+            time (DateTime): the time during the day
+
+        Returns:
+            int
+        """
+        return self.j_business_schedule.businessTimeElapsed(time)
+
+
+class BusinessCalendar(JObjectWrapper):
+    """ A business calendar. """
+
+    j_object_type = _JBusinessCalendar
+
+    def __init__(self, name: str = None):
+        """ Creates a business calendar.
+
+        Args:
+            name (str) : name of the calendar, default is None, which means to use the default Deephaven calendar
```

Comment:
Note that this is more complex than just adding the `Calendar` methods to `BusinessCalendar`, since it is possible for users to create non-business calendars -- if they really want to.